### DISCUSSION TITLE
Newbie question: GO fails to see my 4 keyboards, detecting only 2 ones

### DISCUSSION THREAD

**Discussion by larspalo** [GrandOrgue/grandorgue#430] [Help]
Originally posted 2014-03-14 20:14:04.679000 by organeriamidi in GrandOrgue discussion forum on SourceForge<br><br>Hello, friends.

My setup consists of 3 Keystation 61es and a midi-pedalboard, all connected to a Mac Mini (OSX Lion 10.7.5) through USB. Firstly I connected the four devices through a Hub, and the Midi Devices tab showed, as input devices, only the following ones:
- USB Keystation 61es
- USB2.0 - MIDI port 1

Therefore I was able to use the &quot;Listen to Event&quot; feature to activate only one keyboard and the pedalboard.

I reconnected the 3 keyboards directly to different USB ports in the Mac Mini, and the pedalboard through the Hub to the 4th USB port, but GO continues seeing the same as above.

Thanks in advance for any insight.
Rafael

**Comment by larspalo** [comment]
Originally posted 2014-03-14 21:12:41.773000 by [Martin Koegler](https://github.com/e9925248) in GrandOrgue discussion forum on SourceForge<br><br>GO recognizes MIDI devices by their device name (reported by OS X). From your description I guess, that OS X reports the same ID for all keyboards - so they are regarded as one by GO. You could try changing the MIDI channels of the keyboards - maybe it will help.

PS: GO on Linux is able to handle such a setup out of the box.

**Reply by larspalo** [reply]
Originally posted 2014-03-14 21:58:24.556000 by organeriamidi in GrandOrgue discussion forum on SourceForge<br><br>Thanks, Martin, I just tried your suggestion of changing the MIDI channels, but it didn&#039;t work.

**Comment by larspalo** [comment]
Originally posted 2014-03-15 06:55:40.167000 by [Lars Palo](https://github.com/larspalo) in GrandOrgue discussion forum on SourceForge<br><br>@e9925248; I remember vaguely that there was some fiddeling with the RtMidi (?) code to get the reporting correctly both in Linux and Windows. Might be worth checking if the same code exist for Mac (at the time I last checked it no Mac build existed of GO).

@OP: Are you sure that if you set the MIDI keyboards to different channels and manually configure the onscreen keyboards to respond to the different channels of your MIDI keyboards that it won&#039;t work? Is it so that if you right click on an element and then hit the key of any of your MIDI keyboards that only one would trigger a detection? Otherwise manually configuring the channels should work. However, the set MIDI channels of the keystations are not kept if you shut them down. A better solution would perhaps be using something like a ESI M4U XT 4-in/4-out with standard MIDI cables to workaround the issue in a stable way.

Kind regards

Lars P

**Comment by larspalo** [comment]
Originally posted 2014-03-15 17:33:37.814000 by organeriamidi in GrandOrgue discussion forum on SourceForge<br><br>Dear Lars,

Thanks for your kind attention. Effectively I put different channels (1, 2, 3, 4) in the Initial MIDI Configuration tab, then right-click on every keyboard in the GO screen, where I manually match their respective channels to the selected ones. But the Listen to Event feature doesn´t respond for keyboards 3 (swell) and 4 (choir). I guess that whereas these keyboards do not appear listed in the MIDI Input Devices of the MIDI Devices Tab, that procedure won´t fix the issue.

On the other hand I try to avoid investing in MIDI interfaces, which I don´t need for my setup be able to work quite well in Hauptwerk, for example. The mine is a low-budget project.

I could install Ubuntu, but sure we all want that GO works faultlessly under OSX too.

Best regards,
Rafael

**Reply by OomCarl** [reply]
I have the same problem - using Mojave or Catalina. Changing the midi channels manually does not work. Only one keyboard is recognized.

**Reply by rousseldenis** [reply]
@OomCarl Do you have the last version of GrandOrgue from here ?

**Reply by oleg68** [reply]
This is a well-known problem of GrandOrgue on OSx. The root cause is that if several keyboards have the same ID, OSx exposes only one keyboard to applications, so GO can use only one.

There was a soltion to install some external program that recognizes two keyboards at the low level and exposes to two virtual keyboards wit different IDs to OSx so applications (including GrandOrgue) can use both.

**Reply by willeke1234** [reply]
The keyboards have the same name but different IDs. See: Problem with recognizing two identical MIDI keyboards #500.

**Comment by OomCarl** [comment]
Dear OlegThanks for your kind reply, it is much appreciated. I have the newest version of GrandOrgue and the problem persists. I really wish this matter could be addressed as it would be such a good alternative to Hauptwerk. By the way, the problem was also present in Sweelinq. There is no problem with this in Hauptwerk.Kind regardsCarlSent from my iPadOn Jan 5, 2023, at 4:30 AM, Oleg Samarin ***@***.***> wrote:﻿
This is a well-known problem of GrandOrgue on OSx. The reason is that if several keyboards have the same ID, OSx exposes only one keyboard to applications, so GO can use only one.
There was a soltion to install some external program that recognizes two keyboards at the low level and exposes to two virtual keyboards wit different IDs to OSx so applications (including GrandOrgue) can use both.

**Reply by willeke1234** [reply]
Note: I don't have any MIDI USB devices so I'm just guessing.
Do you see all keyboards in the MIDI Studio window of the Audio MIDI Setup app? It's in the Utilities folder in the Apps folder.
